--- src/flac.c	2020-05-19 19:17:08.234706931 +0200
+++ src/flac-fixed.c	2020-05-19 19:33:33.401411107 +0200
@@ -1160,12 +1160,12 @@
 
 	while (--count >= 0)
 	{	scaled_value = src [count] * normfact ;
-		if (CPU_CLIPS_POSITIVE == 0 && scaled_value >= (1.0 * 0x7F))
+		if (scaled_value >= (1.0 * 0x7F))
 		{	dest [count] = 0x7F ;
 			continue ;
 			} ;
-		if (CPU_CLIPS_NEGATIVE == 0 && scaled_value <= (-8.0 * 0x10))
-		{	dest [count] = 0x80 ;
+		if (scaled_value <= (-8.0 * 0x10))
+		{	dest [count] = -0x80 ;
 			continue ;
 			} ;
 		dest [count] = lrintf (scaled_value) ;
@@ -1182,12 +1182,12 @@
 
 	while (--count >= 0)
 	{	scaled_value = src [count] * normfact ;
-		if (CPU_CLIPS_POSITIVE == 0 && scaled_value >= (1.0 * 0x7FFF))
+		if (scaled_value >= (1.0 * 0x7FFF))
 		{	dest [count] = 0x7FFF ;
 			continue ;
 			} ;
-		if (CPU_CLIPS_NEGATIVE == 0 && scaled_value <= (-8.0 * 0x1000))
-		{	dest [count] = 0x8000 ;
+		if (scaled_value <= (-8.0 * 0x1000))
+		{	dest [count] = -0x8000 ;
 			continue ;
 			} ;
 		dest [count] = lrintf (scaled_value) ;
@@ -1202,13 +1202,13 @@
 
 	while (--count >= 0)
 	{	scaled_value = src [count] * normfact ;
-		if (CPU_CLIPS_POSITIVE == 0 && scaled_value >= (1.0 * 0x7FFFFF))
+		if (scaled_value >= (1.0 * 0x7FFFFF))
 		{	dest [count] = 0x7FFFFF ;
 			continue ;
 			} ;
 
-		if (CPU_CLIPS_NEGATIVE == 0 && scaled_value <= (-8.0 * 0x100000))
-		{	dest [count] = 0x800000 ;
+		if (scaled_value <= (-8.0 * 0x100000))
+                {	dest [count] = -0x800000 ;
 			continue ;
 			}
 		dest [count] = lrintf (scaled_value) ;
@@ -1291,12 +1291,12 @@
 
 	while (--count >= 0)
 	{	scaled_value = src [count] * normfact ;
-		if (CPU_CLIPS_POSITIVE == 0 && scaled_value >= (1.0 * 0x7F))
+		if (scaled_value >= (1.0 * 0x7F))
 		{	dest [count] = 0x7F ;
 			continue ;
 			} ;
-		if (CPU_CLIPS_NEGATIVE == 0 && scaled_value <= (-8.0 * 0x10))
-		{	dest [count] = 0x80 ;
+		if (scaled_value <= (-8.0 * 0x10))
+		{	dest [count] = -0x80 ;
 			continue ;
 			} ;
 		dest [count] = lrint (scaled_value) ;
@@ -1313,12 +1313,12 @@
 
 	while (--count >= 0)
 	{	scaled_value = src [count] * normfact ;
-		if (CPU_CLIPS_POSITIVE == 0 && scaled_value >= (1.0 * 0x7FFF))
+		if (scaled_value >= (1.0 * 0x7FFF))
 		{	dest [count] = 0x7FFF ;
 			continue ;
 			} ;
-		if (CPU_CLIPS_NEGATIVE == 0 && scaled_value <= (-8.0 * 0x1000))
-		{	dest [count] = 0x8000 ;
+		if (scaled_value <= (-8.0 * 0x1000))
+		{	dest [count] = -0x8000 ;
 			continue ;
 			} ;
 		dest [count] = lrint (scaled_value) ;
@@ -1335,12 +1335,12 @@
 
 	while (--count >= 0)
 	{	scaled_value = src [count] * normfact ;
-		if (CPU_CLIPS_POSITIVE == 0 && scaled_value >= (1.0 * 0x7FFFFF))
+		if (scaled_value >= (1.0 * 0x7FFFFF))
 		{	dest [count] = 0x7FFFFF ;
 			continue ;
 			} ;
-		if (CPU_CLIPS_NEGATIVE == 0 && scaled_value <= (-8.0 * 0x100000))
-		{	dest [count] = 0x800000 ;
+		if (scaled_value <= (-8.0 * 0x100000))
+		{	dest [count] = -0x800000 ;
 			continue ;
 			} ;
 		dest [count] = lrint (scaled_value) ;
